src/Icons/src/Registry/LocalSvgIconRegistry.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons\Registry;
+
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\Finder\Finder;
+use Symfony\UX\Icons\Exception\IconNotFoundException;
+use Symfony\UX\Icons\IconRegistryInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class LocalSvgIconRegistry implements IconRegistryInterface
+{
+    public function __construct(private string $iconDir)
+    {
+    }
+
+    public function get(string $name): array
+    {
+        if (!file_exists($filename = sprintf('%s/%s.svg', $this->iconDir, str_replace(':', '/', $name)))) {
+            throw new IconNotFoundException(sprintf('The icon "%s" (%s) does not exist.', $name, $filename));
+        }
+
+        $svg = file_get_contents($filename) ?: throw new \RuntimeException(sprintf('The icon file "%s" could not be read.', $filename));
+        $crawler = (new Crawler($svg))->filter('svg');
+        $node = $crawler->getNode(0) ?? throw new \RuntimeException(sprintf('The icon file "%s" is not a valid SVG.', $filename));
+        $attributes = [];
+
+        if ($viewBox = $node->attributes?->getNamedItem('viewbox')?->nodeValue) {
+            $attributes['viewBox'] = $viewBox;
```

Comment:
Only comment for tonight (brace yourself for tomorrow 😅)

We need a solution to keep some (or add some) or some svg packages won't be usable without adding a lot of manual things in every call :| 

I'm thinking height and width in particular

Reply:
Doesn't viewbox cover height and width? I've tested with several different icons for separate collections and didn't see any issues.

Reply:
Height and view are set to zero if not given, so the icon won't be displayed without additionnal CSS

And it's a performance front-end best practices.

That's an example but my point is: i don't think we should, i'll do more tests tomorrow and give you other examples (lucide with their path strok / round instructions is another example)

Reply:
Documentation (if you want to digg) https://html.spec.whatwg.org/multipage/embedded-content-other.html#dimension-attributes

Reply:
For now, I've changed to not remove any attributes

Reply:
How much time do you let me to suggest an implementation that would solve this question ?

Reply:
I guess I just want svg icons downloaded from _some icon registry site_ to _just work_ without any extra requirements. I thought there was some issues with height/width attributes taking precedence over css classes but as you've explained, that's not the case.